### ISSUE TITLE
Pin.meta query string encoding

### ISSUE THREAD

**Issue by sanderpick** [ipfs/go-pinning-service-http-client#8]
When listing pins, `Pin.meta` is sent as `fmt.Sprintf("%v", obj)` where `obj` is `map[string]string`, e.g., `meta=map[foo:one bar:two]` (after URL unescaping). That's a bit awkward to parse on the backend. Is there a more idiomatic way to send a `map` as a query param? 

Gin [encodes maps](https://github.com/gin-gonic/gin#map-as-querystring-or-postform-parameters) by exploding the keys: "meta[foo]=one&meta[bar]=two".

If I generate the client with another popular Go OpenAPI generator called [oapi-codegen](https://github.com/deepmap/oapi-codegen), it uses [this lib](https://github.com/deepmap/oapi-codegen/blob/master/pkg/runtime/styleparam.go) to encode `Pin.meta` as URL escaped JSON.

**Comment by lidel** [comment]
I am not aware of anyone using meta field yet, (go-ipfs does not use it), so appreciate you looking at this :+1: 

Quick pointers if someone has time to investigate this:

- The spec at https://ipfs.github.io/pinning-services-api-spec/ does not provide clear guideline, but examples like  `meta={"app_id":"99986338-1113-4706-8302-4420da6158aa"}` meant to suggest that `Pin.meta` is passed as (escaped) JSON

- If go-pinning-service-http-client does something else, then it can be:
  - a bug in generator, which needs to be fixed upstream
     (future-proof fix for  `openapi/` dir replaced during client code generation)
  - OR/AND a lack of specificity in the YAML spec that need to be clarified
    (for example, maybe we need to use [explicit content: application/json](https://swagger.io/docs/specification/describing-parameters/#schema-vs-content) for PinMeta)

**Comment by sanderpick** [comment]
Cool, thanks @lidel. I dug into this a bit more last week but spaced pinging back. The openapi generator naively `fmt.Sprintf("%v", obj)`s all param types... but they seem to be [open to tweaks](https://github.com/OpenAPITools/openapi-generator/pull/325/files). I'd be happy to submit a PR there, but need to know how you all would like the data to be encoded (JSON, exploded map, etc.).

**Comment by lidel** [comment]
@sanderpick I don't think we need to change anything in the YAML used for generating the client (we are already use  [suggested YAML for JSON object in query params](https://swagger.io/docs/specification/describing-parameters/#schema-vs-content)), but I proposed spec clarification in the included docs https://github.com/ipfs/pinning-services-api-spec/pull/74 to make it explicit we expect `meta` to be sent as URL-encoded stringified JSON object when passed  in a query param:

- `?meta=%7B%22app_id%22%3A%22UUID%22%7D`

lmk if this is enough for you to fix openapi-generator for golang.

**Comment by sanderpick** [comment]
Cool! I'll get a PR submitted tomorrow.
